### ISSUE TITLE
Can't build openssl 1.1.1a with VS2005 on windows 7

### ISSUE THREAD

**Issue by hvbang** [openssl/openssl#8074]
hi all,
I buildding openssl on windows with config :
I installed ActivePerl-5.24.3.2404-MSWin32-x64-404865.exe, nasm-2.14.03rc2-installer-x86 and download source nasm-2.14.03rc2-installer-x86.
I run cmd with permission Admin and added localtion of ActivePerl to %patch% :

```
@set path=%path%;C:\Program Files (x86)\NASM;C:\Perl64\site\bin;C:\Perl64\bin;
call "c:\Program Files (x86)\Microsoft Visual Studio 8\VC\bin\vcvars32.bat";
```

and run command on folder openssl:
```
perl Configure VC-WIN32 no-shared no-asm --prefix=C:/openssl
nmake
======================================
I have error :
crypto\ct\ct_prn.c(38) : error C2220: warning treated as error - no 'object' file generated
crypto\ct\ct_prn.c(38) : warning C4244: 'function' : conversion from 'uint64_t' to 'long', possible
loss of data
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 8\VC\BIN\cl.EXE"' : return code '0x2'
Stop.

NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio 8\VC\BIN\nmake.exe"' : return code '0x2'
Stop.
```

Please help me to be able to build successfully !
Thank you very much !

**Comment by mspncp** [comment]
The warning seems to be caused by the last argument to `ASN1_GENERALIZEDTIME_ad()`.

```C
        ASN1_GENERALIZEDTIME *ASN1_GENERALIZEDTIME_adj(ASN1_GENERALIZEDTIME *s,
                                                       time_t t, int offset_day,
                                                       long offset_sec);
```

```C											   
		ASN1_GENERALIZEDTIME_adj(gen, (time_t)0,
                             (int)(timestamp / 86400000),
                             (long)(timestamp % 86400000) / 1000);
```

Im not sure, but maybe it's because your compiler is relatively old? It is possible (but I don't know) that modern compilers notice the fact that the `(timestamp % 86400000)` expression fits into a `long` value (because 86400000 < 2^31 = 2147483648) and don't complain, even though `timestamp` is declared as `uint64_t` and `long` is a 32bit type. IMHO it should be safe to add `(long)` cast to the last argument as follows.
													   

Please let us know whether this helps. But apart of that you might run into different problems because of your old compiler. If possible, please consider upgrading.

**Comment by hvbang** [comment]
Sorry, I clicked delete history comment.
the current, my code is 

> 
ASN1_GENERALIZEDTIME_adj(gen, (time_t)0,
                             (int)(timestamp / 86400000),
                             (timestamp % 86400000) / 1000);
> 
but this error still happen

**Comment by levitte** [comment]
Wild idea: how about `86400000LL`?

**Comment by hvbang** [comment]
hi @levitte,
the error still happen :(

**Comment by yutakakn** [comment]
Hi @hvbang 

Please refer to #8948 issue.
I succeeded that OpenSSL 1.1.1b was built with VS2005.

**Comment by levitte** [comment]
To solve the "warning treated as error" problem, we need to backport #10287 to 1.1.1

**Comment by mathewhodson** [comment]
Pull requests #10287 and #10365 were closed, so I think this is fixed.

**Comment by nhorman** [comment]
Marking as inactive, to be closed at the end of 3.4 dev barring further input
